### PR TITLE
DDCE-1001: Added allowed Country codes functionality to Charity's off…

### PR DESCRIPTION
**Pull request by robert-webb-hmrc**
…ical address lookup

# DDCE-1001

##New feature

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/connectors/addressLookup/AddressLookupConfiguration.scala [hunk 1]
```diff
@@ -65,4 +65,45 @@ class AddressLookupConfiguration @Inject()(callbackUrl: String,
     )
   }
 
+  def applyEEA: AddressLookupEEAConfigurationModel = {
+    val english = Lang("en")
+    val welsh = Lang("cy")
+
+    AddressLookupEEAConfigurationModel(
```
Comment (abhijeetardale) [inline app/connectors/addressLookup/AddressLookupConfiguration.scala:72]: Can we use AddressLookupOptionsModel model with last parameter as
allowedCountryCodes: Option[Set[String]] = None

File: app/connectors/addressLookup/AddressLookupConfiguration.scala [hunk 2]
```diff
@@ -65,4 +65,45 @@ class AddressLookupConfiguration @Inject()(callbackUrl: String,
...
+      version = 2,
+      options = AddressLookupEEAOptionsModel(
+        continueUrl = SafeRedirectUrl(appConfig.host + callbackUrl),
+        phaseFeedbackLink = SafeRedirectUrl(appConfig.feedbackUrlAddressLookup),
+        signOutHref = SafeRedirectUrl(controllers.routes.SignOutController.signOut().url),
+        deskProServiceName = appConfig.contactFormServiceIdentifier,
+        showPhaseBanner = true,
+        showBackButtons = true,
+        includeHMRCBranding = false,
+        allowedCountryCodes =
```
Comment (abhijeetardale) [inline app/connectors/addressLookup/AddressLookupConfiguration.scala:82]: should be optional set

File: app/models/addressLookup/AddressLookupEEAConfigurationModel.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.addressLookup
+
+import play.api.libs.json.{Json, Writes}
+
+case class AddressLookupEEAConfigurationModel(version: Int,
```
Comment (abhijeetardale) [inline app/models/addressLookup/AddressLookupEEAConfigurationModel.scala:21]: not required, please use same model

File: app/models/addressLookup/AddressLookupEEAOptionsModel.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * limitations under the License.
+ */
+
+package models.addressLookup
+
+import play.api.libs.json.{Json, Writes}
+import uk.gov.hmrc.play.bootstrap.binders.SafeRedirectUrl
+
+case class AddressLookupEEAOptionsModel(continueUrl: SafeRedirectUrl,
+                                        phaseFeedbackLink: SafeRedirectUrl,
```
Comment (abhijeetardale) [inline app/models/addressLookup/AddressLookupEEAOptionsModel.scala:23]: not required please use same model

File: app/connectors/addressLookup/AddressLookupConnector.scala [hunk 5]
```diff
@@ -38,6 +38,13 @@ class AddressLookupConnector @Inject()(httpClient: HttpClient, implicit val appC
     )
   }
 
+  def initializeEEA(callbackUrl: String, messagePrefix: String, fullName: Option[String])
+                (implicit hc: HeaderCarrier, ec: ExecutionContext, messages: MessagesApi): Future[AddressLookupInitializationResponse] = {
+    httpClient.POST(addressLookupInitUrl, new AddressLookupConfiguration(callbackUrl, messagePrefix, fullName).applyEEA)(
+      AddressLookupEEAConfigurationModel.writes, AddressLookupInitializationReads, hc, ec
```
Comment (abhijeetardale) [inline app/connectors/addressLookup/AddressLookupConnector.scala:44]: please use same model writes

**Review by abhijeetardale — CHANGES_REQUESTED** [review]

**Comment by robert-webb-hmrc** [comment]
Updated

**Review by abhijeetardale — CHANGES_REQUESTED** [review]
could you please check below for Czech Republic, its showing not found however Czechia is present in list which is another name for Czech Republic
image

**Comment by robert-webb-hmrc** [comment]
Based on the country code Json this is not something we can change as the CZ country code will return Czechia as the country name. This will need further discussion as Czechia is the official name that country is known to the European union as of 2016 http://publications.europa.eu/code/en/en-370100.htm

**Review by abhijeetardale — APPROVED** [review]

**Pull request merged**